### DISCUSSION TITLE
Allow sources to be configured to take an api-key even if they don't require one

### DISCUSSION THREAD

**Discussion by HeafieldLuke** [projectdiscovery/subfinder#885] [Ideas]
## The Problem
The AlienVault source does not allow the ability to provide an API key, which means that the AlienVault source is constrained by the 1,000 requests per hour rate limit that the AlienVault anonymous API provides.

A free-tier API key allows for 10,000 requests per hour, which would be nice to take advantage of, and would allow a higher-tier API key to make even more requests per hour.

I was looking at making a contribution for this feature, and figured I would start a discussion thread here as looking through the available sources, it looks like they are all set-up for either having an api key, or using an anonymous api (if I'm missing one, please feel free to point me to it). It's easy enough to modify the AlienVault source to need an api-key, but not sure if there's precedent for sources that don't require api keys to then require an one.

**Comment by ehsandeep** [comment]
@tarunKoyalwar see this, might be related to your work on [optimization-code-refactoring](https://github.com/projectdiscovery/subfinder/tree/optimization-code-refactoring) branch.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
